.circleci/config.yml

```diff
@@ -10,6 +10,23 @@ refs:
         else
           echo "no skip_job file present, full steam ahead"
         fi
+  skip_server_tests: &skip_server_tests
+    run:
+      name: checking if is PR build & server changes not present
+      command: |
+        echo "PR number: $CIRCLE_PR_NUMBER"
+        if [ -z "$CIRCLE_PR_NUMBER" ]; then
+          echo "Not a PR build, continue test" && exit 0
+        fi
+        echo "Compare refs: ${CIRCLE_COMPARE_URL##http*/}" 
+        for folder in `git log --format="" --name-only ${CIRCLE_COMPARE_URL##http*/} | cut -d"/" -f1 | sort -u`; do
+          echo "contents in $folder were modified"
+          if [ "$folder" = "server" ]; then
+            echo "Server changes present, continue test" && exit 0
+          fi
+        done
```

Comment:
This seems rather complicated. I think this would do the same thing in a much more direct way:

```sh
if git diff --quiet "${CIRCLE_COMPARE_URL##http*/}" -- server; then
  echo "No server changes present, skipping test"
  circleci-agent step halt
else
  echo "Server changes present, continuing test"
fi
```

More generally: this `CIRCLE_COMPARE_URL` environment variable is mysterious! Looking at the job output, it appears to be empty according to the “Preparing Environment Variables” step, but clearly it is not. Searching the internet for information proves somewhat unhelpful, but it seems to work, so I guess that’s fine.

Reply:
Yes. Even the CircleCI docs don't seem to mention much about it other than its existence. But like you said it seems to work.

Will update the PR with your suggestions.

Reply:
Done